src/main/java/com/client/core/base/tools/entitychanger/impl/StandardEntityChanger.java

```diff
@@ -0,0 +1,148 @@
+                throw new RuntimeException(e);
+            }
+        }, (entity1, entity2) -> entity1);
+        try {
+            Method getter = target.getClass().getDeclaredMethod(getterMethodName(finalField));
+            Class<?> fieldType = getter.getReturnType();
+            Method setter = target.getClass().getDeclaredMethod(setterMethodName(finalField), fieldType);
+            if (value instanceof Map) {
+                if (getter.invoke(target) == null) {
+                    setter.invoke(target, fieldType.newInstance());
+                }
+
+                T nestedTarget = (T) getter.invoke(target);
+                ((Map<String, ?>) value).forEach((key, mapVal) -> {
+                    try {
+                        setField(nestedTarget, key, mapVal);
+                    } catch (RuntimeException e) {
+                        log.error("Error setting nested field " + key, e);
+                    }
+                });
+            } else {
+                try {
+                    setter.invoke(target, value);
+                } catch (IllegalAccessException | InvocationTargetException | IllegalArgumentException e) {
+                    if (DateTime.class.equals(fieldType)) {
+                        setter.invoke(target, asType(value, DateTime.class));
+                    } else if (BullhornEntity.class.isAssignableFrom(fieldType)) {
+                        BullhornEntity bullhornEntity = (BullhornEntity) fieldType.newInstance();
+                        bullhornEntity.setId((Integer) value);
+                        setter.invoke(target, bullhornEntity);
+                    } else if (Integer.class.equals(fieldType)) {
+                        setter.invoke(target, asType(value, Integer.class));
+                    } else if (BigDecimal.class.equals(fieldType)) {
+                        setter.invoke(target, asType(value, BigDecimal.class));
+                    } else if (Boolean.class.equals(fieldType)) {
+                        setter.invoke(target, asType(value, Boolean.class));
+                    } else {
+                        log.error("Error setting field " + finalField + " to value " + value, e);
+                    }
+                }
+            }
+        } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException |
+                 InstantiationException e) {
+            log.error("Error setting field " + finalField + " to value " + value + ", no property " + finalField + " on " + target.getClass().getSimpleName(), e);
+        }
+        return entity;
+    }
+
+    private Object getValue(Object entity, String field) {
+        return Arrays.stream(field.split("\\.")).reduce(entity, (entityOrField, nextField) -> {
+            try {
+                Method getter = entityOrField.getClass().getDeclaredMethod(getterMethodName(nextField));
+                return getter.invoke(ObjectUtils.defaultIfNull(entityOrField, entityOrField.getClass().newInstance()));
```

Comment:
This `newInstance()` has been deprecated since java 9. You can use the following to achieve the same results
```suggestion
                return getter.invoke(ObjectUtils.defaultIfNull(entityOrField, entityOrField.getClass().getDeclaredConstructor().newInstance()));
```

Reply:
There's a couple of instances of this method in this file